src/System.Net.Http/src/System/Net/Http/Unix/CurlHandler.MultiAgent.cs

```diff
@@ -559,14 +559,11 @@ private static size_t CurlReceiveHeadersCallback(IntPtr buffer, size_t size, siz
 
                         if (!TryParseStatusLine(response, responseHeader, easy))
                         {
-                            int colonIndex = responseHeader.IndexOf(':');
-
-                            // Skip malformed header lines that are missing the colon character.
-                            if (colonIndex > 0)
+                            int index = 0;
+                            string headerName = CurlResponseParseUtils.ReadHeaderName(responseHeader, out index);
+                            if (headerName != null)
                             {
-                                string headerName = responseHeader.Substring(0, colonIndex);
-                                string headerValue = responseHeader.SubstringTrim(colonIndex + 1);
-
+                                string headerValue = responseHeader.Substring(index + 1).Trim();
```

Comment:
> I was also referring to the increment in ReadHeaderName itself.

Are you referring to [```index++;```](https://github.com/dotnet/corefx/pull/4131/files#diff-8ed24cdb38638a3b114dd5a1f040a9d3R118)?

That is correct and safe; it reaches that line only when [```index < responseHeader.Length```](https://github.com/dotnet/corefx/pull/4131/files#diff-8ed24cdb38638a3b114dd5a1f040a9d3R114). 
After the increment, index can at the most be equal to ```responseHeader.Length```. And ```x.Substring(x.Length)``` returns empty string and does not throw.



>  if all remaining chars are spaces, headerValue will get set to string.Empty - I am assuming that is ok

Yes. empty header values are allowed, from HTTP point of view. 

If you prefer, I can add a Unit test around this area. please let me know.